### PR TITLE
Updating logic (along with Dave) to permit Merchant install at initial UI usage CH-33138

### PR DESCRIPTION
**Pull request by CTDoriott**
[Ticket](https://app.clubhouse.io/ns8/story/33138/trigger-merchant-record-creation-upon-ui-view)

This change permits Merchant installation during initial UI visit

### CODE REVIEW

File: etc/events.xml [hunk 1]
```diff
@@ -12,4 +12,7 @@
   <event name="adminhtml_block_html_before">
     <observer name="ns8_sales_order_view_update" instance="NS8\Protect\Observer\Order\ViewUpdate" />
   </event>
+  <event name="ns8_protect_dashboard_container_instantiated">
+    <observer name="ns8_protect_dashboard_merchant_installation" instance="NS8\Protect\Observer\Adminhtml\DashboardInstantiation" />
```
Comment (davehensleyns8) [inline etc/events.xml:16]: Shouldn't this be in `...\Admin\`?

File: Observer/Admin/DashboardInstantiation.php [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+{
+    /**
+     * Registry key used to determine fetching the access token between setup and update
+     */
+    const ACCESS_TOKEN_SET_KEY = 'ns8_access_token_set';
+
+    /**
+     * The  config manager
+     *
+     * @var Config $scopeConfig;
```
Comment (davehensleyns8) [inline Observer/Admin/DashboardInstantiation.php:31]: Technically we don't need the variable name on these (especially when it's wrong). 😄

File: Setup/Uninstall.php [hunk 3]
```diff
@@ -48,6 +48,7 @@ public function uninstall(SchemaSetupInterface $setup, ModuleContextInterface $c
         try {
             $setup->startSetup();
             UninstallerClient::uninstall();
+            $this->config->setIsMerchantActive(false);
```
Comment (davehensleyns8) [inline Setup/Uninstall.php:51]: CI found a bug... if the user installs the module but never goes to the dashboard afterward, they will get this error when they try to uninstall: `An authorization token is required for NS8 requests`

Comment (CTDoriott) [reply Setup/Uninstall.php:55]: This has been addressed in the latest commit

**Pull request merged**